blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/AggregatedPayload.java

```diff
@@ -113,6 +113,10 @@ private static void validate(AggregatedPayload payload) {
                 metricName = ((BluefloodSet) leafBean).getName();
             }
 
+            if ((metricName.isEmpty()) && (payload.getAllMetricNames().size() > 0)) {
+                metricName = payload.getAllMetricNames().get(0);
+            }
```

Comment:
Yes, I tested this.  Each "single" aggregatedPayload will have the metricName of the offending metric as the first element.  And actually, it'll always be the case that each instance of AggregatedPayload will only contain one leaf "metric".  Here's an example of the part of a payload this class represent:

```
{
      "tenantId":"333333",
      "timestamp":1478711535000,
      "flushInterval": 15000,
      "counters":[
          {
              "name":"agg.count.ten",
              "value":1,
              "rate":1
          }
      ]
    }
```

Even if the metric above is part of a multi-metrics payload like this:
```
curl -i -H "content-type: application/json" -X POST 'http://localhost:2440/v2.0/836986/ingest/aggregated/multi' -d \
  '[
    {
      "tenantId":"333333",
      "timestamp":1478711535000,
      "flushInterval": 15000,
      "counters":[
          {
              "name":"agg.count.nine",
              "value":1,
              "rate":1
          }
      ]
    },
    {
      "tenantId":"333333",
      "timestamp":1478711535000,
      "flushInterval": 15000,
      "counters":[
          {
              "name":"agg.count.ten",
              "value":2,
              "rate":1
          }
      ]
    },
    {
      "tenantId":"333333",
      "timestamp":1478711535000,
      "flushInterval": 15000,
      "counters":[
          {
              "name":"agg.count.eleven",
              "value":1,
              "rate":1
          },
          {
              "name":"agg.count.twelve",
              "value":1,
              "rate":1
          }
      ]
    }
  ]'
```

This is because if you look in the HttpAggregatedMultiIngestionHandler class, it first parse the entire json payload to each individual element (stored in bundleList: 
https://github.com/rackerlabs/blueflood/blob/master/blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpAggregatedMultiIngestionHandler.java#L76
and
https://github.com/rackerlabs/blueflood/blob/master/blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpAggregatedMultiIngestionHandler.java#L176)

Then for each of the element in bundleList (an AggregatedPayload), it does the validation and create the error payload if there's violation:  https://github.com/rackerlabs/blueflood/blob/master/blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpAggregatedMultiIngestionHandler.java#L85


In the majority of the cases, the metricName is pulled from the previous block in the AggregatedPayload.validate() here:  https://github.com/rackerlabs/blueflood/blob/master/blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/AggregatedPayload.java#L105-L114 

with the assumption that leafBean was created correctly from the json parser.  

However, there were cases where the parser itself didn't create the AggregatedPayload element correctly (e.g. a missing "value" field that is required), resulting in the type of the element not being set and an metricName.  In this case, the only way to retrieve the metricName is through the single json element payload itself. 

Hope that make sense.  But do know that I tried testing different cases including one with multiple aggregated payloads, with the first few being valid and the last one being invalid, and the error response has the correct metricName for the one failing the validation (and not ingested).  You'll see a few of this cases in the system integration tests itself.